### PR TITLE
RFC - Part 3

### PR DESCRIPTION
**Pull request by kris-daxiom**
*Issue:*

https://github.com/bcgov/entity/issues/22386
https://github.com/bcgov/entity/issues/22347
https://github.com/bcgov/entity/issues/22352

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Review by dimak1 — APPROVED** [review]
Looks good to me.

File: strr-api/src/strr_api/services/application_service.py [hunk 1]
```diff
@@ -109,18 +120,53 @@ def update_application_payment_details_and_status(application: Application, invo
...
+                event_name=Events.EventName.PAYMENT_COMPLETE,
+                application_id=application.id,
+            )
+        return application
+
+    @staticmethod
+    def update_application_status(application: Application, application_status: Application.Status, reviewer: User):
+        """Updates the application status. If the application status is approved, a new registration is created."""
+        application.status = application_status
+        if application.status in [Application.Status.APPROVED, Application.Status.REJECTED]:
```
Comment (shaangill025) [inline strr-api/src/strr_api/services/application_service.py:163]: Maybe use `APPLICATION_TERMINAL_STATES`?

File: strr-api/src/strr_api/services/application_service.py [hunk 2]
```diff
@@ -109,18 +120,53 @@ def update_application_payment_details_and_status(application: Application, invo
...
+                event_type=Events.EventType.APPLICATION,
+                event_name=Events.EventName.APPLICATION_SUBMITTED,
+                application_id=application.id,
+            )
         else:
-            if application.status == Application.Status.DRAFT:
-                application.status = Application.Status.SUBMITTED
+            if application.payment_status_code == PaymentStatus.COMPLETED.value:
+                application.status = Application.Status.PAID
+                application.payment_completion_date = datetime.fromisoformat(invoice_details.get("paymentDate"))
```
Comment (shaangill025) [inline strr-api/src/strr_api/services/application_service.py:143]: Can you clarify why this block is needed?

**Review by shaangill025 — APPROVED** [review]
LGTM

**Pull request merged**